app/src/main/java/com/android/sample/model/profile/ProfileViewModel.kt

```diff
@@ -1,17 +1,22 @@
 package com.android.sample.model.profile
 
-import android.util.Log
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.android.sample.model.activity.database.AppDatabase
 import com.google.firebase.Firebase
 import com.google.firebase.auth.auth
 import dagger.hilt.android.lifecycle.HiltViewModel
 import javax.inject.Inject
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
 
 @HiltViewModel
-open class ProfileViewModel @Inject constructor(private val repository: ProfilesRepository) :
```

Comment:
Injecting AppDatabase into ProfileViewModel to access local database operations.